### PR TITLE
fix(rum-core): handle script error events properly

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ fixes https://github.com/elastic/apm-agent-rum-js/issues/416
+ When script error is thrown, error key is set to `null` which passes the typeof `object` check. We handle this case in this PR and the code portions are merged to make it cleaner.

### CODE REVIEW

File: packages/rum-core/test/error-logging/error-logging.spec.js [hunk 1]
```diff
@@ -220,7 +220,8 @@ describe('ErrorLogging', function() {
       colno
     })
     listener('error', {
-      message: 'Script error.' + testErrorMessage
+      message: 'Script error.' + testErrorMessage,
+      error: null
```
Comment (jahtalab) [inline packages/rum-core/test/error-logging/error-logging.spec.js]: Shouldn't we also test for undefined as well as null?

File: packages/rum-core/src/error-logging/error-logging.js [hunk 2]
```diff
@@ -48,32 +48,34 @@ class ErrorLogging {
       culprit = lastFrame.filename
     }
 
-    const message =
-      errorEvent.message || (errorEvent.error && errorEvent.error.message)
```
Comment (jahtalab) [inline packages/rum-core/src/error-logging/error-logging.js]: The logic here prioritises `errorEvent.message` over `errorEvent.error.message` if they both exists! The change reverses this priority, is that intentional? What's the reasoning?

File: packages/rum-core/test/error-logging/error-logging.spec.js (hunk 1 above)
Comment (vigneshshanmugam) [reply packages/rum-core/test/error-logging/error-logging.spec.js]: Its always set to null during the creation https://www.w3.org/TR/html50/webappapis.html#dom-errorevent-error

File: packages/rum-core/src/error-logging/error-logging.js (hunk 2 above)
Comment (vigneshshanmugam) [reply packages/rum-core/src/error-logging/error-logging.js]: I didn't get you here. 

https://github.com/elastic/apm-agent-rum-js/blob/cd916b88395b879b6b472b8733d63fbe2938ef90/packages/rum-core/src/error-logging/error-logging.js#L52-L59

We set it to `errorEvent.message` first at line 52 and set it to `errorEvent.error.message` later on line 56.

File: packages/rum-core/test/error-logging/error-logging.spec.js (hunk 1 above)
Comment (jahtalab) [reply packages/rum-core/test/error-logging/error-logging.spec.js]: I vaguely remember that some browsers have/had a different behaviour even though the spec says it needs to be null!

File: packages/rum-core/src/error-logging/error-logging.js (hunk 2 above)
Comment (jahtalab) [reply packages/rum-core/src/error-logging/error-logging.js]: Consider this:

```
let errorEvent = {message:'message', error: {message:'error-message'}}
```

The previous logic will result in `errorMessage` to be `message` since `errorEvent.message || (errorEvent.error && errorEvent.error.message)` only checks the first part of the OR operator and never checks the `error`. While the current changes will set `errorMessage` to `error-message`

Comment (vigneshshanmugam) [reply packages/rum-core/src/error-logging/error-logging.js]: Got it now, It was intentional that my impression that both messages are similar, I was wrong here. I will update and also add test case for the same.
